sketch/README.md

```diff
@@ -0,0 +1,243 @@
+var pdf; // store all the parameters of a mixture-density distribution
+var temperature = 0.45; // controls the amount of uncertainty of the model
+var line_color;
+var model_loaded = false;
+
+// loads the TensorFlow.js version of sketch-rnn model, with the "cat" model's weights.
+model = new ms.SketchRNN("https://storage.googleapis.com/quickdraw-models/sketchRNN/models/cat.gen.json");
+// code that ensures the above line is run before the below lines are run.
+
+function setup() {
+  x = windowWidth/2.0;
+  y = windowHeight/3.0;
+  createCanvas(windowWidth, windowHeight);
+  frameRate(60);
+
+  // initialize the scale factor for the model. Bigger -> large outputs
+  model.set_pixel_factor(3.0);
+
+  // initialize pen's states to zero.
+  [dx, dy, pen_down, pen_up, pen_end] = model.zero_input(); // the pen's states
+
+  // zero out the rnn's initial states
+  rnn_state = model.zero_state();
+
+  // define color of line
+  line_color = color(random(64, 224), random(64, 224), random(64, 224));
+};
+
+function draw() {
+
+  // see if we finished drawing
+  if (prev_pen[2] == 1) {
+    noLoop(); // stop drawing
+    return;
+  }
+
+  // using the previous pen states, and hidden state, get next hidden state
+  // the below line takes the most CPU power, especially for large models.
+  rnn_state = model.update([dx, dy, pen_down, pen_up, pen_end], rnn_state);
+
+  // get the parameters of the probability distribution (pdf) from hidden state
+  pdf = model.getPDF(rnn_state, temperature);
+
+  // sample the next pen's states from our probability distribution
+  [dx, dy, pen_down, pen_up, pen_end] = model.sample(pdf);
+
+  // only draw on the paper if the pen is touching the paper
+  if (prev_pen[0] == 1) {
+    stroke(line_color);
+    strokeWeight(3.0);
+    line(x, y, x+dx, y+dy); // draw line connecting prev point to current point.
+  }
+
+  // update the absolute coordinates from the offsets
+  x += dx;
+  y += dy;
+
+  // update the previous pen's state to the current one we just sampled
+  prev_pen = [pen_down, pen_up, pen_end];
+};
+```
+
+## Train own model
+
+There is a small IPython [notebook](https://github.com/tensorflow/magenta-demos/blob/master/jupyter-notebooks/Sketch_RNN_TF_To_JS_Tutorial.ipynb) to show how to quickly train a sketch-rnn model with Python-based TensorFlow model, and convert that model over to the JSON format that can be used by by this model.
+
+## Demos
+
+A few demos are available in `demos` directory built to use the Sketch RNN model.  You can look at the corresponding code to study in detail how the model works.  To run these examples, it is recommended to use a simple local webserver, such as the http-server that can be obtained using npm, or the `python -m http.server` command, and load the local html file from the local server.  Some examples require this, since they need to dynamically load .json model files, and local static session doesn't allow for this in many browsers.
```

Comment:
you can also launch a server with yarn, see `yarn run-demos` in music

Reply:
Makes sense, I'll check out `yarn run-demos` and amend the doc.